src/Microsoft.OData.Core/UriParser/Parsers/UriQueryExpressionParser.cs

```diff
@@ -284,9 +284,19 @@ internal AggregateExpressionToken ParseAggregateExpression()
         {
             // expression
             var expression = this.ParseExpression();
+            var endPathExpression = expression as EndPathToken;
+            AggregationMethod verb;
 
-            // "with" verb
-            var verb = this.ParseAggregateWith();
+            if (endPathExpression != null && endPathExpression.Identifier == ExpressionConstants.QueryOptionCount)
+            {
+                expression = new EndPathToken("$count", endPathExpression.NextToken);

```

Comment:
Isn't new EndPathToken() redundant?  Why does expression not already == EndPathToken("$count", endPathExpression.NextToken)?